frontera/core/models.py

```diff
@@ -23,10 +24,7 @@ def __init__(self, url, method='GET', headers=None, cookies=None, meta=None, bod
         :param dict cookies: dictionary of cookies to attach to this request.
         :param dict meta: dictionary that contains arbitrary metadata for this request.
         """
-        self._url = url
-        self._method = str(method).upper()
-        self._url = to_native_str(url, encoding)
-        self._encoding = encoding
+        self._url = safe_url_string(url, encoding)
```

Comment:
Ideal would be to add additional field for storing parsed URL. Because many components are performing url parsing, so they could make use of such structure.